libs/community/langchain_community/document_loaders/blob_loaders/file_system.py

```diff
@@ -58,6 +68,7 @@ def __init__(
         glob: str = "**/[!.]*",
         exclude: Sequence[str] = (),
         suffixes: Optional[Sequence[str]] = None,
+        metadata_dict: Optional[Dict[str, Dict[str, Any]]] = None,
```

Comment:
This logic should be handled outside the file system blob loader. The code that is using the BlobLoader can inspect the the blobs and add metadata to them based on the path associated with them.